### ISSUE TITLE
WalletConnect not working for Polygon bridge

### ISSUE THREAD

**Issue by AW-STJ** [AlphaWallet/alpha-wallet-ios#2846]
Not sure why, but the app cannot connect to Polygon bridge.

https://wallet.matic.network/bridge/

![image](https://user-images.githubusercontent.com/61957841/120605189-22517d80-c491-11eb-84a7-9d69e6ee4f04.png)

**Comment by hboon** [comment]
It looks like they didn't integrate a recent version of the WalletConnect SDK so it doesn't support mobile linking — i.e when you visit the page within the dapp browser or in mobile Safari on iOS, it doesn't that that is happening and bring up the list of wallets to choose from. i.e it doesn't do what this example does: https://example.walletconnect.org. Don't think we can fix this for our end. If the user visits the page from another browser (probably desktop), WalletConnect should work properly.

Here's what I get if I visit the dapp on Safari desktop and scan the QR code:

<img width="200" alt="Screenshot 2021-06-03 at 5 36 33 PM" src="https://user-images.githubusercontent.com/56189/120623548-65f8b700-c492-11eb-96ef-5d2e91d8fceb.png">
<img width="200" alt="Screenshot 2021-06-03 at 5 36 42 PM" src="https://user-images.githubusercontent.com/56189/120623570-6b560180-c492-11eb-8036-6855803cbcbf.png">

It prompts to sign a message right away and once we get past that, the dapp *seems* to load. I didn't try to do anything with it. So let me know if the dapp itself doesn't work for us.

**Comment by zhangzhongnan928** [comment]
If you have a DApp browser on your mobile, you should never use WalletConnect to connect a DApp on your mobile and your DApp browser.

The bridge can inject our web3 directly, but won't pop up TX action page. It just keeps loading. 

May be we should create an issue on their repo. 

![image](https://user-images.githubusercontent.com/33795543/120629355-140f5b80-c4a9-11eb-8af6-0bf8f316d7f8.png)

**Comment by hboon** [comment]
Are you referring to this problem? https://github.com/AlphaWallet/alpha-wallet-ios/issues/2839#issuecomment-853750139 Seems to work for me

**Comment by zhangzhongnan928** [comment]
> Are you referring to this problem? https://github.com/AlphaWallet/alpha-wallet-ios/issues/2839#issuecomment-853750139 Seems to work for me

Yes, but I didn't use WalletConnect, I connect through web3 browser option.

**Comment by AW-STJ** [comment]
>  I connect through web3 browser option.

I was able to connect through the clicking the Metamask option on our Dapp browser. 
![image](https://user-images.githubusercontent.com/61957841/120636518-12499600-c4b1-11eb-9980-0b4c957dcf37.png)

>  never use WalletConnect to connect a DApp on your mobile

Good learning, as I do use WalletConnect sometimes to connect to Dapps if they do not connect automatically. 

@colourfreak - can you raise this issue with Polygon, as you are already connected with them ?

**Comment by hboon** [comment]
Ah. Was checking both this and #2839 and was confused. Just to be clear: #2839 is when accessed with the dapp browser directly. And for this issue, which is for connecting with WalletConnect, it seems to work/not work as expected, according to https://github.com/AlphaWallet/alpha-wallet-ios/issues/2846#issuecomment-853736045
